src/main/java/io/confluent/connect/elasticsearch/DataConverter.java

```diff
@@ -200,11 +202,19 @@ static Object preProcessValue(Object value, Schema schema, Schema newSchema) {
         }
         return result;
       case MAP:
-        keySchema = schema.keySchema();
-        valueSchema = schema.valueSchema();
-        ArrayList<Struct> mapStructs = new ArrayList<>();
-        Map<?, ?> map = (Map<?, ?>) value;
+        Schema keySchema = schema.keySchema();
         Schema newValueSchema = newSchema.valueSchema();
+        Schema valueSchema = schema.valueSchema();
```

Comment:
nit: Since we declare a variable that is used only once below (that is: `Schema valueSchema = schema.valueSchema();`) let's add `Schema newKeySchema = newSchema.keySchema()` as well. 
Also I suggest that they'd be added in an order that feels more consistent (old first then new ones, or key schemas first then value schemas next, again old then new).